application-admintools-default/src/main/java/com/xwiki/admintools/internal/util/DefaultFileOperations.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.InstantiationStrategy;
+import org.xwiki.component.descriptor.ComponentInstantiationStrategy;
+
+/**
+ * Collection of file operations.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component(roles = DefaultFileOperations.class)
+@InstantiationStrategy(ComponentInstantiationStrategy.PER_LOOKUP)
+public class DefaultFileOperations
```

Comment:
Why do you need a class for this? (it's probably one of those that you added for the test error and will be deleted)

Reply:
I did not find a way around this class. I still need it for the tests because Scanner class is final.

Reply:
Maybe you can try other options, to see if it's the same? Check `BufferedReader`

Reply:
I did try using BufferedReader, but I cannot mock File.exists(). I searched on the internet and could not find a solution

Reply:
I think you could create a temporary file from your test https://github.com/xwiki/xwiki-platform/blob/3686040943a3d82ae2c62784c0a9bcae8b9f430b/xwiki-platform-core/xwiki-platform-store/xwiki-platform-store-filesystem-oldcore/src/test/java/org/xwiki/store/filesystem/internal/WikiDeletedListenerTest.java#L60